### PR TITLE
Turing sampler docs

### PR DESCRIPTION
**Pull request by denainjs**
I started writing a documentation page called How Turing samplers implement AbstractMCMC. It takes the [Interface Guide](https://turing.ml/dev/docs/for-developers/interface) as a prerequisite, and gives an overview of the structs and functions used in `src/inference/` files, taking Importance Sampling as an example. 

Among other things, it covers:

- the links and hierarchy between samplers, sampling methods, states, instances of `VarInfo`, etc.
- how `assume` and `observe` are used during sampling

I followed the instructions [here](https://github.com/denainjs/Turing.jl/tree/turing_sampler_docs/docs) in order to add the file `how_turing_implements_abstractmcmc.md` in `Turing.jl/docs/src/for-developers`, and to build the site locally. However, I am not able to access the new page with my browser, as I get the following error messages:

- `ERROR '/versions.js' not found`
- `ERROR '/dev/docs/for-developers/how_turing_implements_abstractmcmc' not found`

### CODE REVIEW

**Comment by devmotion** [comment]
Great, every improvement of the documentation is helpful IMO! However, I guess you only want to add the markdown file but not all other generated files?

BTW we should make sure to update the documentation after switching to https://github.com/TuringLang/AbstractMCMC.jl/pull/42.

**Comment by devmotion** [comment]
See also https://github.com/TuringLang/AbstractMCMC.jl/pull/43#issuecomment-652924221.

**Comment by yebai** [comment]
@denainjs is this ready for a review?

File: docs/.jekyll-cache/Jekyll/Cache/Jekyll--Cache/b7/9606fb3afea5bd1609ed40b622142f1c98125abcfe89a76a661b0e8e343910 [hunk 1]
```diff
@@ -0,0 +1 @@
...am"}, "values"=>{"layout"=>"team", "sidebar"=>false}}, {"scope"=>{"path"=>"_posts", "type"=>"posts"}, "values"=>{"layout"=>"post", "sidebar"=>false}}], "liquid"=>{"error_mode"=>"warn", "strict_filters"=>false, "strict_variables"=>false}, "kramdown"=>{"auto_ids"=>true, "toc_levels"=>[1, 2, 3, 4, 5, 6], "entity_output"=>"as_char", "smart_quotes"=>"lsquo,rsquo,ldquo,rdquo", "input"=>"GFM", "hard_wrap"=>false, "guess_lang"=>true, "footnote_nr"=>1, "show_warnings"=>false}, "title"=>"Turing.jl", "email"=>"support@turing.jl", "author"=>"The Turing Team", "description"=>"Turing: A robust, efficient and modular library for general-purpose probabilistic programming.\n", "url"=>"http://localhost:4000", "repo"=>"https://github.com/TuringLang/Turing.jl", "github_user"=>"TuringLang", "github_repo"=>"Turing.jl", "twitter"=>"Turingjl", "linkedin"=>"Turingjl", "dockerhub"=>"Turingjl", "logo"=>"assets/img/favicon.ico", "logo_pixels"=>34, "color"=>"#d2e6f5", "navbarcolor"=>"#073c44", "tag_search_endpoint"=>"https://ask.cyberinfrastructure.org/search?q=", "tag_color"=>"danger", "accentColor"=>"red", "themeColor"=>"red", "fixedNav"=>"true", "livereload_port"=>35729, "serving"=>true, "watch"=>true}:ET
```
Comment (yebai) [inline docs/.jekyll-cache/Jekyll/Cache/Jekyll--Cache/b7/9606fb3afea5bd1609ed40b622142f1c98125abcfe89a76a661b0e8e343910:1]: @denainjs Perhaps consider removing these immediate files from Jekyll?

File: docs/src/for-developers/how_turing_implements_abstractmcmc.md [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+alg = IS()
+```
+
+defines an Importance Sampling algorithm, an instance of `IS` which is a subtype of `InferenceAlgorithm`. 
+
+When creating your own Turing sampling method, you must therefore build a subtype of `InferenceAlgorithm` corresponding to your method.
+
+### Samplers
+
+Samplers are **not** the same as algorithms. An algorithm is a generic sampling method, a sampler is an object that stores information about how algorithm and model interact during sampling, and is modified as sampling progresses. The `Sampler` struct is defined in DynamicPPL.
```
Comment (yebai) [inline docs/src/for-developers/how_turing_implements_abstractmcmc.md:94]: nice explanation!

File: docs/.jekyll-cache/Jekyll/Cache/Jekyll--Cache/b7/9606fb3afea5bd1609ed40b622142f1c98125abcfe89a76a661b0e8e343910 (hunk 1 above)
Comment (cpfiffer) [reply docs/.jekyll-cache/Jekyll/Cache/Jekyll--Cache/b7/9606fb3afea5bd1609ed40b622142f1c98125abcfe89a76a661b0e8e343910:1]: Not sure why `.jekyll-cache` was checked in. It should be removed from the PR and we should probably add `.jekyll-cache` to the `.gitignore` file.

File: docs/src/contributing/guide.md [hunk 3]
```diff
@@ -1,6 +1,3 @@
----
```
Comment (yebai) [inline docs/src/contributing/guide.md:1]: We need to keep these `yaml` headers, I think. Similarly for all other deletions below. 

Maybe @cpfiffer can confirm whether we need these.

Comment (cpfiffer) [reply docs/src/contributing/guide.md:1]: Yeah, these should all be left in.

**Pull request merged**